tableauserverclient/models/property_decorators.py

```diff
@@ -46,3 +46,34 @@ def wrapper(self, value):
         return func(self, value)
 
     return wrapper
+
+
+def property_is_valid_time(func):
+    @wraps(func)
+    def wrapper(self, value):
+        units_of_time = {"hour", "minute", "second"}
+
+        if not any(hasattr(value, unit) for unit in units_of_time):
+            error = "Invalid time object defined."
+            raise ValueError(error)
+        return func(self, value)
+
+    return wrapper
+
+
+def property_is_int(min, max=2**31):
+    def property_type_decorator(func):
+        @wraps(func)
+        def wrapper(self, value):
+            if min is None:
+                return func(self, value)
```

Comment:
I don't get this? Why can't I set the min to None and the max to some number (like 100). Not saying you are doing this right now but this code would break. Just do the if "min is not None and value < min" and similarly "if max is not None and value > max". No?